src/FileModule.cc

```diff
@@ -180,6 +181,16 @@ AbstractNode *FileModule::instantiateWithFileContext(FileContext *ctx, const Mod
 		auto instantiatednodes = this->scope.instantiateChildren(ctx);
 		node->children.insert(node->children.end(), instantiatednodes.begin(), instantiatednodes.end());
 	}
+	catch (AssertionFailedException &e) {
+		auto docPath = boost::filesystem::path( ctx->documentPath() );
+		auto uncPath = boostfs_uncomplete((*(e.loc.filePath())), docPath);
+
+		std::stringstream msg;
+		msg << e.what();
+		msg << " failed in file " << uncPath.generic_string() <<",";
+		msg << " line " << e.loc.firstLine();
+		PRINT(msg.str());
```

Comment:
Using `PRINTB` may make this look nicer by being able to drop the `stringstream`. I'll let you decide what works best.

Reply:
I see your point, but... that could go wrong if not done properly. Let us unwind it: msg << e.what(); is the "old"/current message inc. "ERROR  Assertion " followed by the expression... and the expression could contain %s.

I could off-course move the whole message building into the exception handler and use %s for including the expression... but going down that rabbit hole ends in rewriting all error messages incl. making them translatable. So, that sounds like rainy holiday project. I´ll make my self a note.

Reply:
Not sure I understand that concern. This is not a bit deal, but wouldn't this work?
```
PRINTB("%s failed in file %s, line %d", e.what(), uncPath.generic_string(), e.loc.firstLine());
```

Reply:
Ah - now I see, I will do that.